pkg/drivers/libhvee/libhvee_windows.go

```diff
@@ -0,0 +1,309 @@
+	return "libhvee"
+}
+
+func (d *Driver) GetState() (state.State, error) {
+	log.Debugf("Machine: libhvee -> state")
+	vm, err := d.getMachine()
+	if err != nil {
+		return state.Error, err
+	}
+
+	log.Debugf("Machine: libhvee -> state: get")
+	vmState := vm.State()
+	switch vmState {
+	case hypervctl.Enabled:
+		log.Debugf("Machine: libhvee -> state: running")
+		return state.Running, nil
+	case hypervctl.Disabled:
+		log.Debugf("Machine: libhvee -> state: stopped")
+		return state.Stopped, nil
+	}
+
+	log.Debugf("Machine: libhvee -> state: unknown")
+	return state.Error, fmt.Errorf("Unknown state")
+}
+
+// PreCreateCheck checks that the machine creation process can be started safely.
+func (d *Driver) PreCreateCheck() error {
+	// Check that powershell was found
+	if _, err := exec.LookPath("powershell.exe"); err != nil {
+		return ErrPowerShellNotFound
+	}
+
+	// Check that hyperv is installed
+	if err := hypervAvailable(); err != nil {
+		return err
+	}
+
+	// Check that the user is an Administrator
+	isAdmin, err := isAdministrator()
+	if err != nil {
+		return err
+	}
+	if !isAdmin {
+		return ErrNotAdministrator
+	}
+
+	return nil
+}
+
+func (d *Driver) Create() error {
+	log.Debugf("Machine: libhvee -> creating: system settings")
+	systemSettings, err := hypervctl.NewSystemSettingsBuilder().
+		PrepareSystemSettings(d.MachineName, nil).
+		PrepareMemorySettings(
+			func(ms *hypervctl.MemorySettings) {
+				ms.DynamicMemoryEnabled = d.DynamicMemory
+				ms.VirtualQuantity = uint64(d.Memory)
+				ms.Reservation = 1024
+				ms.Limit = uint64(d.Memory)
+			}).
+		PrepareProcessorSettings(
+			func(ps *hypervctl.ProcessorSettings) {
+				ps.VirtualQuantity = uint64(d.CPU)
+			}).
+		Build()
+
+	if err != nil {
+		return err
+	}
+
+	log.Debugf("Machine: libhvee -> creating: copy disk image")
+	diskPath := d.getDiskPath()
+	if err := crcos.CopyFile(d.ImageSourcePath, diskPath); err != nil {
+		return err
+	}
+
+	log.Debugf("Machine: libhvee -> creating: hardware setup")
+	err = hypervctl.NewDriveSettingsBuilder(systemSettings).
+		AddScsiController().
+		AddSyntheticDiskDrive(0).
+		DefineVirtualHardDisk(diskPath,
+			func(vhdss *hypervctl.VirtualHardDiskStorageSettings) {}).
+		Finish().
+		Finish().
+		Finish().
+		Complete()
+
+	if err != nil {
+		return err
+	}
+
+	log.Debugf("Machine: libhvee -> creating: done")
+
+	return nil
+}
+
+func (d *Driver) getMachine() (*hypervctl.VirtualMachine, error) {
+	vmm := hypervctl.NewVirtualMachineManager()
+
+	log.Debugf("Machine: libhvee -> get machine")
+	return vmm.GetMachine(d.MachineName)
+}
+
+// Start starts an host
+func (d *Driver) Start() error {
+	vm, err := d.getMachine()
+	if err != nil {
+		return err
+	}
+
+	log.Debugf("Machine: libhvee -> start")
+	return vm.Start()
+}
+
+// Stop stops an host
+func (d *Driver) Stop() error {
+	vm, err := d.getMachine()
+	if err != nil {
+		return err
+	}
+
+	log.Debugf("Machine: libhvee -> stop")
+	return vm.Stop()
```

Comment:
Is this synchronous until the vm state is stopped? Or do we need to keep the `waitStopped` helper from the hyperv code?

Reply:
libhvee implementation is synchronous: https://github.com/containers/libhvee/blob/0698042e8c753a95c349a2c45f8a6b04a5aa2e3a/pkg/hypervctl/vm.go#L266-L276